tests/integration/deregister-trailer.int.test.ts

```diff
@@ -0,0 +1,53 @@
+import supertest from 'supertest';
+import { emptyDatabase, populateDatabase } from '../utils/dbUtil';
+import * as domain from '../../src/domain';
+
+const url = 'http://localhost:3020/local';
+const request = supertest(url);
+
+describe('deregister trailer', () => {
+  beforeAll(async () => {
+    await emptyDatabase();
+  });
+
+  beforeEach(async () => {
+    await populateDatabase();
+  });
+
+  afterEach(async () => {
+    await emptyDatabase();
+  });
+
+  afterAll(async () => {
+    await populateDatabase();
+  });
+
+  test('Unregister when TRN is found', async () => {
```

Comment:
we should also check the behaviour when the trailer has previously been deregistered, and when it isn't found in the database

Reply:
Following tests added:
- when trn is found
- when trn already exists
- when payload is invalid
- when trn is not found